src/components/SearchResults.tsx

```diff
@@ -28,9 +29,35 @@ export default function SearchResults(props: SearchResultsProps) {
     });
   },[searchQuery])
 
+  // function for handeling arrow down - arrow up 
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === 'ArrowDown' || e.key === 'ArrowUp') {
+      e.preventDefault();
+      if (resultsRef.current) {
```

Comment:
Considering checking for the negative here and returning early if this reference is not set..

```js
if (!resultsRef.current) return;
```

Then the code currently wrapped in this if statement can just continue normally.